Objective-C/CBLReplicator.mm

```diff
@@ -550,11 +550,14 @@ - (bool) handleError: (C4Error)c4err {
         CBLLogInfo(Sync, @"%@: Transient error (%@); will retry in %.0f sec...",
                    self, error.localizedDescription, delay);
         [self scheduleRetry: delay];
-    } else {
+    } else if (!_reachability.reachable) {
```

Comment:
is there any possibility that reachability can be turned back on, when the control reaches here? which means it will stay disconnected.

Reply:
Permanent error when not come here. It should be already returned.